plugins/item_previews/web_client/views/ItemPreviewWidget.js

```diff
@@ -1,103 +1,180 @@
-        return /\.(jpg|jpeg|png|gif)$/i.test(item.get('name'));
+    setElement: function ($el) {
+        this.initialized = false;
+        return View.prototype.setElement.apply(this, arguments);
     },
 
     render: function () {
-        var supportedItems = this.collection.filter(this._isSupportedItem.bind(this));
-
-        var view = this;
-
-        view.$el.html(ItemPreviewWidgetTemplate({
-            items: supportedItems,
-            wrapPreviews: view.wrapPreviews,
-            hasMore: this.collection.hasNextPage(),
-            isImageItem: this._isImageItem,
-            isJSONItem: this._isJSONItem
-        }));
-
-        // Render any JSON files.
-        supportedItems.filter(this._isJSONItem).forEach(function (item) {
-            var id = item.get('_id');
-
-            // Don't process JSON files that are too big to preview.
-            var size = item.get('size');
-            if (size > this._MAX_JSON_SIZE) {
-                return view.$('.json[data-id="' + id + '"]').text('JSON too big to preview.');
-            }
+        if (!this.initialized) {
+            this.initialized = true;
+            this.renderedIndex = 0;
+            this.$el.html(ItemPreviewWidgetTemplate());
+            this.$('.g-widget-item-previews-container').on('scroll', (e) => this.onScroll(e));
+            this.debouncedaddMoreItems();
+        }
+    },
+
+    isNotFull: function () {
+        return this.$('.g-widget-item-previews-wrapper').height() <= this.$('.g-widget-item-previews-container').height();
+    },
+
+    allItemAdded: function () {
+        return this.renderedIndex >= this.supportedItems.length;
+    },
+
+    addMoreItems: function () {
+        this.addingMoreItems = true;
+        var items = this.supportedItems.slice(this.renderedIndex, this.renderedIndex + this._LOAD_BATCH_SIZE);
+        this.renderedIndex = Math.min(this.renderedIndex + this._LOAD_BATCH_SIZE, this.supportedItems.length);
+
+        if (!items.length) {
+            return;
+        }
 
-            // Ajax request the JSON files to display them.
-            restRequest({
-                path: 'item/' + id + '/download',
-                type: 'GET',
-                error: null // don't do default error behavior (validation may fail)
-            }).done(function (resp) {
-                view.$('.json[data-id="' + id + '"]').text(JSON.stringify(resp, null, '\t'));
-            }).fail(function (err) {
-                console.error('Could not preview item', err);
+        var $container = this.$('.g-widget-item-previews-wrapper');
+        _whenAll(items.map((item) => {
+            return restRequest({
+                path: `item/${item._id}/files`,
+                data: { limit: 5 }
+            }).then((files) => {
+                return _whenAll(files.map((file) => {
+                    return this.tryGetFileContent(file, file.mimeType);
+                }));
+            });
+        })).then((results) => {
```

Comment:
You can simplify this by using `.done(` instead of `.then(` here, and removing the `return undefined;` at the end of the function.